advancedFunctions/decorator.js

```diff
@@ -0,0 +1,18 @@
+const tribonacci = (signature,n) => {
+    const arr = signature;
+    for(let count = 3; count < n; count ++) {
+        arr.push(arr.slice(-3).reduce((a, b) => a + b));
+      }
+    arr.slice(0, n).forEach((element) => console.log(element));
+  }
+
+const slower = (func, seconds) => {
+  const wrapper = (...args) => {
+    console.log(`Chill out bro, you will get you result in ${seconds} seconds`);
+    return setTimeout(func, seconds*1000, ...args);
```

Comment:
this is good that you know the third param in the setTimeout but let's use the `apply` or `call` methods.

Reply:
Done